### DISCUSSION TITLE
ScriptReference/PropertyName

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#35] [Notes]
# ScriptReference/PropertyName

https://docs.unity3d.com/ScriptReference/PropertyName.html

**Comment by vertxxyz** [comment]
If you're using PropertyNames for Reference Values in [PlayableDirector](https://docs.unity3d.com/ScriptReference/Playables.PlayableDirector.html) bindings (ie. [ExposedReference](https://docs.unity3d.com/ScriptReference/ExposedReference_1.html) in TimelineClip) the usage of this type is to:
- Create a new PropertyName with your own generated GUID.
- Assign the PropertyName to your [`exposedName`](https://docs.unity3d.com/ScriptReference/ExposedReference_1-exposedName.html).
- [`SetReferenceValue`](https://docs.unity3d.com/ScriptReference/Playables.PlayableDirector.SetReferenceValue.html) with that PropertyName and your binding.

If you attempt to use the original exposedName it will be `:0`, an empty PropertyName. This value will then be used for all unexposed properties, and that's definitely not something you want to be doing!
